src/node_worker.cc

```diff
@@ -644,7 +644,37 @@ void Worker::StartThread(const FunctionCallbackInfo<Value>& args) {
           w->JoinThread();
           // implicitly delete w
         });
-  }, static_cast<void*>(w)), 0);
+  }, static_cast<void*>(w));
+  if (ret != 0) {
+    char err_buf[128];
+    uv_err_name_r(ret, err_buf, sizeof(err_buf));
+    w->custom_error_ = "ERR_WORKER_INIT_FAILED";
+    w->custom_error_str_ = err_buf;
+    w->loop_init_failed_ = true;
+    w->thread_joined_ = true;
+    w->stopped_ = true;
+    w->env()->remove_sub_worker_context(w);
+    {
+      HandleScope handle_scope(w->env()->isolate());
+      Context::Scope context_scope(w->env()->context());
+      // Reset the parent port as we're closing it now anyway.
+      w->object()->Set(w->env()->context(),
+                    w->env()->message_port_string(),
+                    Undefined(w->env()->isolate())).Check();
```

Comment:
Please return here if the result of `Set()` is empty instead of using `.Check()`.